app/uk/gov/hmrc/exports/movements/services/NotificationService.scala

```
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.movements.services
+
+import javax.inject.{Inject, Singleton}
+import play.api.Logger
+import uk.gov.hmrc.exports.movements.models.notifications.MovementNotification
+import uk.gov.hmrc.exports.movements.repositories.NotificationRepository
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class NotificationService @Inject()(notificationRepository: NotificationRepository) {
+
+  private val logger = Logger(this.getClass)
+
+  def save(notification: MovementNotification): Future[Either[String, Unit]] =
+    notificationRepository
+      .insert(notification)
+      .map(_ => Right((): Unit))
+      .recover {
```

Why not let it bubble up to the top? It saves us having to do `Left` and `Right` handling everywhere for exceptional circumstances

---

In some cases when we have exceptions thrown, we don't want the client service to know about the problems, so we just log the problem but still return Accepted status code.
This is also how we do it on exports BE, so I wanted to be concise.